plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/MultipleSelectionAction.java

```diff
@@ -0,0 +1,166 @@
+ */
+public abstract class MultipleSelectionAction extends Action implements IEventListener {
+
+	private final ISelectionChangedListener listener = new ISelectionChangedListener() {
+		@Override
+		public void selectionChanged(SelectionChangedEvent event) {
+			final IStructuredSelection selection = (IStructuredSelection) event.getSelection();
+			selectionElementChanged(isValidSelection(selection));
+		}
+	};
+	
+	Object viewer;
+	protected boolean connectionSelected;
+	protected IFeature[] featureArray;
+
+	/**
+	 * Default constructor
+	 * @param text of the action in context menu
+	 * @param viewer2
+	 */
+	public MultipleSelectionAction(String text, Object viewer2) {
+		super(text);
+		viewer = viewer2;
+		setEnabled(false);
+		if (viewer2 instanceof GraphicalViewerImpl) {
+			((GraphicalViewerImpl) viewer2).addSelectionChangedListener(listener);
+		} else {
+			((TreeViewer) viewer2).addSelectionChangedListener(listener);
+		}
+	}
+	
+	/**
+	 * returns the selected features as IFeatures
+	 * @return selected IFeature array
+	 */
+	protected IFeature[] getSelectedFeatures() {
+		ArrayList<IFeature> features = new ArrayList<>();
+		
+		IStructuredSelection selection;
+		if (viewer instanceof TreeViewer) {
+			selection = (IStructuredSelection) ((TreeViewer) viewer).getSelection();
+			if (selection.getFirstElement() instanceof FmOutlineGroupStateStorage) {
+				for (Object obj : selection.toArray()) {
+					features.add(((FmOutlineGroupStateStorage) obj).getFeature());
+				}
+				return features.toArray(new IFeature[features.size()]);
+			} else {
+				for (Object obj : selection.toArray()) {
+					features.add((IFeature)obj);
+				}
+				return features.toArray(new IFeature[features.size()]);
+			}
+		} else {
+			selection = (IStructuredSelection) ((AbstractEditPartViewer) viewer).getSelection();
+		}
+
+		final Object part = selection.getFirstElement();
+		connectionSelected = part instanceof ConnectionEditPart;
+		if (connectionSelected) {
+			for (Object obj : selection.toArray()) {
+				IFeature tempFeature = ((ConnectionEditPart) obj).getModel().getTarget().getObject();
+				features.add(tempFeature);
+			}
+			return features.toArray(new IFeature[features.size()]);
+		}
+		for (Object obj : selection.toArray()) {
+			features.add(((FeatureEditPart) obj).getModel().getObject());
+		}
+		return features.toArray(new IFeature[features.size()]);
+	}
+	
+	/**
+	 * Is called whenever the selection changes
+	 * @param validSelection
+	 */
+	protected void selectionElementChanged(boolean validSelection) {
+		if (featureArray != null) {
+			for (Object obj : featureArray) {
+				((IFeature)obj).removeListener(this);
+			}
+		}
+		if (validSelection) {			
+			featureArray =  getSelectedFeatures();
+			for (IFeature tempFeature : featureArray) {
+				tempFeature.addListener(this);
+			}
+			updateProperties();
+		} else {
+			featureArray = null;
+			setEnabled(false);
+		}
+	}
+	
+	protected abstract void updateProperties();
```

Comment:
There should be some kind of method header here

Reply:
fixed